### DISCUSSION TITLE
Extracting the dates from entries this page

### DISCUSSION THREAD

**Discussion by gingerbeardman** [FreshRSS/FreshRSS#5546] [Q&A] [answered]
Using HTML + XPath, what would be the recommended way of extracting the dates from entries on this page?

https://www.jordanmechner.com/en/library/1993-journals/

- they're formatted long dates (which I can extract fine) 
- but some have extra text in brackets after them (which is problematic)

Everything else was very easy to extract!

**Comment by gingerbeardman** [comment]
Tried the trick in this issue...

https://github.com/FreshRSS/FreshRSS/issues/5467#issuecomment-1589885020

...but doesn't seem to be working as all articles are dated today?

**Comment by Alkarex** [comment, marked as answer]
Hello,
So you have dates like `<h3>January 11, 1993 [San Francisco]</h3>` or like `<h3>January 12, 1993</h3>`.

You can play with [XPath string methods](https://developer.mozilla.org/docs/Web/XPath/Functions/substring-before). For instance something like:

```
substring-before(concat(descendant::h3, '['), '[')
```

The trick is to concat the character your are looking for, to make sure there is at least one instance of that character.

Similar: https://github.com/FreshRSS/FreshRSS/issues/5467#issuecomment-1589885020

**Reply by gingerbeardman** [reply]
Perfect, thanks! I had tried and failed with that approach. Yours works!
